### PR TITLE
Adding Options bag for parameters 

### PR DESCRIPTION
**Pull request by hamshavathimunibyraiah**
Adding Options bag for API parameters in ListDocumentStatuses and ListTranslationStatuses. Upating tests, samples and readme.

### CODE REVIEW

File: sdk/translation/azure-ai-translation-document/src/main/java/com/azure/ai/translation/document/models/ListDocumentStatusesOptions.java [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+     * 
+     * 
+     * Clients MAY request server-driven paging with a specific page size by
+     * specifying a maxpagesize preference. The server SHOULD honor this preference
+     * if the specified page size is smaller than the server's default page size.
+     * 
+     * @param maxpagesize the maxpagesize value to set.
+     * @return the ListDocumentStatusesOptions object itself.
+     */
+    public ListDocumentStatusesOptions setMaxpagesize(Integer maxpagesize) {
```
Comment (srnagar) [inline sdk/translation/azure-ai-translation-document/src/main/java/com/azure/ai/translation/document/models/ListDocumentStatusesOptions.java:232]: Use camel-casing - `setMaxPageSize()`

File: sdk/translation/azure-ai-translation-document/src/main/java/com/azure/ai/translation/document/models/ListDocumentStatusesOptions.java [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+        this.statuses = statuses == null ? null : new ArrayList<>(statuses);
+        return this;
+    }
+
+    /**
+     * Get the createdDateTimeUtcStart property: the start datetime to get items after.
+     * 
+     * @return the createdDateTimeUtcStart value.
+     */
+    public OffsetDateTime getCreatedDateTimeUtcStart() {
```
Comment (srnagar) [inline sdk/translation/azure-ai-translation-document/src/main/java/com/azure/ai/translation/document/models/ListDocumentStatusesOptions.java:282]: Use `getCreatedAfter()` and `getCreatedBefore()` names to indicate the list fetches items that are created after/before the given datetime.

**Review by srnagar — APPROVED** [review]

File: sdk/translation/azure-ai-translation-document/src/main/java/com/azure/ai/translation/document/models/ListDocumentStatusesOptions.java (hunk 1 above)
Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-document/src/main/java/com/azure/ai/translation/document/models/ListDocumentStatusesOptions.java]: Updated

File: sdk/translation/azure-ai-translation-document/src/main/java/com/azure/ai/translation/document/models/ListDocumentStatusesOptions.java (hunk 2 above)
Comment (hamshavathimunibyraiah) [reply sdk/translation/azure-ai-translation-document/src/main/java/com/azure/ai/translation/document/models/ListDocumentStatusesOptions.java]: Updated

**Pull request merged**